launchable/commands/record/tests.py

```diff
@@ -91,7 +92,7 @@ def tests(context, base_path: str, session: Optional[str], build_name: Optional[
     else:
         session_id = find_or_create_session(
             context, session, build_name, flavor)
-
+        build_name = read_build()
```

Comment:
IIUC, this is problematic.

My first "hmm" was because this overwrites what the user has given. Upon further thoughts, the case where this goes wrong is when we are talking a distributed test and the user uses the `--session` option. We do not expect a context session file in that case, so `read_build()` will return `None`, even if the user specifies some valid value in `--build`. But the real problem in this case is, why would the user have to specify the build name?

What `get_record_start_at` is trying to do is to compute the timestamp of the build. Looking at the logic, it just needs a valid test session ID, which in this context we are guaranteed to have.

So I think the next line should be changed to `record_start_at = get_record-start_at(None, session_id)`